UefiPayloadPkg/UefiPayloadEntry/FitUniversalPayloadEntry.c

```diff
@@ -600,27 +545,58 @@ BuildHobs (
 
 /**
   Entry point to the C language phase of UEFI payload.
-  @param[in]   BootloaderParameter    The starting address of bootloader parameter block.
+  @param[in]   BootloaderParameter  The starting address of FDT .
   @retval      It will not return if SUCCESS, and return error when passing bootloader parameter.
 **/
 EFI_STATUS
 EFIAPI
-_ModuleEntryPoint (
+FitUplEntryPoint (
   IN UINTN  BootloaderParameter
   )
 {
   EFI_STATUS                  Status;
   PHYSICAL_ADDRESS            DxeCoreEntryPoint;
   EFI_PEI_HOB_POINTERS        Hob;
   EFI_FIRMWARE_VOLUME_HEADER  *DxeFv;
+  PHYSICAL_ADDRESS            HobListPtr;
+  VOID                        *FdtBase;
+  VOID                        *FdtBaseResvd;
 
-  mHobList = (VOID *)BootloaderParameter;
-  DxeFv    = NULL;
+  if (FixedPcdGetBool (PcdHandOffFdtEnable)) {
+    mHobList = (VOID *)NULL;
+  } else {
+    mHobList = (VOID *)BootloaderParameter;
+  }
+
+  DxeFv = NULL;
   // Call constructor for all libraries
   ProcessLibraryConstructorList ();
 
   DEBUG ((DEBUG_INFO, "Entering Universal Payload...\n"));
   DEBUG ((DEBUG_INFO, "sizeof(UINTN) = 0x%x\n", sizeof (UINTN)));
+  DEBUG ((DEBUG_INFO, "BootloaderParameter = 0x%x\n", BootloaderParameter));
+
+  DEBUG ((DEBUG_INFO, "Start init Hobs...\n"));
+  HobListPtr = UplInitHob ((VOID *)BootloaderParameter);

```

Comment:
I think we have to add more comments to clarify the flow (it is easy to forget):
In FDT scenario:
1. UplInitHob () will create UPL HobList and update mHobList
2. CustomFdtNodeParser () will get additional hobs from bootloader when FDT/upl-custom present. (HobListPtr)
3. FitBuildHobs () will base on HobListPtr (upl-custom HobList) to add additional Hobs into mHobList (UPL HobList). Hobs created by FDT standard nodes take higher priority than upl-custom Hobs when duplicate Hobs happened.